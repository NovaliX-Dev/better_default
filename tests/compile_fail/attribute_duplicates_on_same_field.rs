#![allow(dead_code)]

use better_default::BetterDefault;

#[derive(BetterDefault)]
enum Enum2 {
    #[default]
    #[default]
    Variant {
        first: u32,
        second: String,
    },

    Variant2,

    Variant3,
}

#[derive(BetterDefault)]
struct Struct {
    #[default(0)]
    #[default(1)]
    field: u32
}

fn main() {}