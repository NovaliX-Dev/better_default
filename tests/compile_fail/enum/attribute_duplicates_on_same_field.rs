#![allow(dead_code)]

use better_default::BetterDefault;

#[derive(BetterDefault)]
enum Enum2 {
    #[default]
    #[default]
    Variant {
        first: u32,
        second: String,
    },

    Variant2,

    Variant3,
}

fn main() {}