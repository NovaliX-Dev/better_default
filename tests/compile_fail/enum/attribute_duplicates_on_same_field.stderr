error: this attribute is already declared.
 --> tests/compile_fail/enum/attribute_duplicates_on_same_field.rs:8:7
  |
8 |     #[default]
  |       ^^^^^^^
